### PR TITLE
Add search box animation

### PR DESCRIPTION
**Pull request by Hitansh-Shah**

### CODE REVIEW

File: apps/core/templates/components/header.html [hunk 1]
```diff
@@ -10,9 +10,26 @@
...
-        <a href="{% url 'search' %}" class="bg-transparent border-0 border-start border-1 border-dark ps-4 ms-4">
+        <button class="bg-transparent border-0 border-start border-1 border-dark ps-4 ms-4" data-bs-toggle="modal" data-bs-target="#searchModal">
           <img src="{% manifest 'images/search.svg' %}" >
-        </a>
+        </button>
+        <div class="modal fade" id="searchModal" tabindex="-1" aria-labelledby="searchModalLabel" aria-hidden="true">
+          <div class="modal-dialog modal-lg">
+            <div class="modal-content rounded-3 fs-6">
+              <div class="modal-body p-4 d-flex align-items-center">
+                <input type="text" placeholder="Search..." class="border border-1 border-dark rounded-start w-75 p-2 ps-3" name="search" />
```
Comment (allcaps) [inline apps/core/templates/components/header.html:20]: Can we keep this a form for now? So it opens the search page on submit?

We can do the live search results in anther PR.

File: apps/core/templates/components/header.html [hunk 2]
```diff
@@ -10,9 +10,26 @@
           <span class="fw-bold me-2">wagtail</span>
```
[the remark this answers is not in the document]
Comment (allcaps) [reply apps/core/templates/components/header.html:10]: Screenshot 2022-07-26 at 15 51 28

- A Needs to be the same distance.
- B:C has some ratio in de design. Now they are equal. I think C need to be bigger.
- Arrow, here the button and input do not align. I think BS should be able to have pixel perfect alignment here.

**Comment by vossisboss** [comment]
Hi @Hitansh-Shah. One small thing I noticed on this PR is that there appears to be a greenish highlight that appears around the search box when you enter text into it (see screenshot). Unless there's an accessibility need for that to be there, I'm thinking we should stick to the original design as much as possible. Or at least make it thinner so it doesn't overlap quite as much with the button.

Screen Shot 2022-08-04 at 3 54 03 PM

**Comment by thibaudcolas** [comment]
@vossisboss this is the field’s focus style to indicate where the user’s focus is. I’ll add a separate task for us to revisit it – it’s actually too subtle right now to pass accessibility checks.

**Comment by allcaps** [comment]
https://developer.mozilla.org/en-US/docs/Web/CSS/:focus#accessibility_concerns

Focus is required for accessibility.

We chose a component framework as this fills in the gaps of the design. We should not fight the defaults of the framework, unless there is something better designed. The framework does the correct thing (have a focus area even if it is not designed).

This focus area is applied to all focusable elements. Try tab navigation. All links will get the same styling.

**Comment by vossisboss** [comment]
I thought there might be an accessibility need there. Thanks for clarifying. This works for me.

**Pull request merged**